prov/usnic/src/usdf_fabric.c

```diff
@@ -623,6 +595,96 @@ usdf_get_distance(
     return ret;
 }
 
+/* Check all things related to a device. Make sure it's okay, the source address
+ * matches the requested address, the destination is reachable from the device,
+ * the device fabric name matches the requested fabric name, and the device
+ * domain name matches the requested domain name.
+ *
+ * @param version Libfabric API version used to verify the domain / fabric name.
+ * @param hints   Hints passed to fi_getinfo.
+ * @param src     Source address being requested.
+ * @param dest    Destination address to communicate with.
+ * @param dep     usNIC device entry being checked.
+ *
+ * @return true on success, false on failure. For debug logging can be enabled
+ *         to see why a device was disqualified.
+ */
+static bool usdf_check_device(uint32_t version, struct fi_info *hints,
+			      struct sockaddr_in *src, struct sockaddr_in *dest,
+			      struct usdf_dev_entry *dep)
+{
+	char dest_str[INET_ADDRSTRLEN];
+	char src_str[INET_ADDRSTRLEN];
+	char dev_str[INET_ADDRSTRLEN];
+	struct usd_device_attrs *dap;
+	int reachable;
+	int ret;
+
+	dap = &dep->ue_dattr;
+
+	/* Skip the device if it has problems. */
+	if (!dep->ue_dev_ok) {
+		USDF_WARN_SYS(FABRIC, "skipping %s/%s\n", dap->uda_devname,
```

Comment:
You might want to add a reason to this string (e.g., ": device is not ok", or somesuch).  This will let a developer/user know that there's something wrong with the device without needing to check the source code of the usnic provider to see that the error message is from the `!dep->ue_dev_ok` check failing.